test/metrics.test.ts

```diff
@@ -1,210 +1,316 @@
-import assert from 'assert';
 const openDigger = require('../src/metrics/index');
+import * as fs from "fs";
+const assert = require("assert");
+import * as _ from "lodash";
 
-describe('Index and metric test', () => {
-  // use repos under google org in 2015 to 2016
-
-  const commonAssert = async (func: (option: any) => Promise<any[]>, key: string, options?: any) => {
-    const years = 2;
-    const months = 24;
-
-    // year group with latest order
-    const option: any = {
-      startYear: 2015, endYear: 2016, startMonth: 1, endMonth: 12,
-      orgIds: [1342004], order: 'DESC', limit: 3,
-      groupTimeRange: 'year',
-      ...(options?.queryOptions ? options.queryOptions : {}),
-    };
-    let result = await func(option);
-    assert.strictEqual(result.length, option.limit); // the limit works fine
-    assert.strictEqual(result.every(r => r[key].length === years), true);  // the main field has correct length
-    // order works fine
-    if (options?.index !== undefined) {
-      assert.strictEqual(parseFloat(result[0][key][years - 1][options.index]) >= parseFloat(result[option.limit - 1][key][years - 1][options.index]), true);
-    } else {
-      assert.strictEqual(parseFloat(result[0][key][years - 1]) >= parseFloat(result[option.limit - 1][key][years - 1]), true);
+function deepEqual(a: any, b: any): boolean {
+  return _.isEqual(a, b);
+}
+
+async function validate_data(
```

Comment:
We should keep identical naming style, the function name should be `validateData`.